srv/modules/runners/proposal.py

```diff
@@ -317,28 +472,37 @@ def populate(**kwargs):
 
     local_client = salt.client.LocalClient()
 
-    proposals = local_client.cmd(args['target'], 'proposal.generate',
-                                 tgt_type='compound', kwarg=args)
-
-    # check if profile of 'name' exists
+    minions_to_replace = []
     profile_dir = '{}/profile-{}'.format(BASE_DIR, args['name'])
-    if not isdir(profile_dir):
-        os.makedirs(profile_dir, 0o755)
-    # TODO do not hardcode cluster name ceph here
-    if not isdir('{}/stack/default/ceph/minions'.format(profile_dir)):
-        os.makedirs('{}/stack/default/ceph/minions'.format(profile_dir), 0o755)
-    if not isdir('{}/cluster'.format(profile_dir)):
-        os.makedirs('{}/cluster'.format(profile_dir), 0o755)
-
-    # determine which proposal to choose
-    for node, proposal in proposals.items():
-        _proposal = _choose_proposal(node, proposal, args)
-        if _proposal:
-            _write_proposal(_proposal, profile_dir)
-    # write out .filter here...will need some logic to merge existing data too.
-    _record_filter(args, profile_dir)
-
-    return True
+    if isdir(profile_dir):
+        minions_to_replace = _find_minions_to_replace(profile_dir)
+
+    if minions_to_replace:
+        for minion in minions_to_replace:
+            ReplaceDiskOn(minion).replace()
+        return True
+    else:
```

Comment:
Does this mean that if a replacement is ongoing no new proposals are created? Wondering if that'll bite us at some point when someone replaces a disk and adds a new node in one go...

Reply:
That is actually a good point I did not think about. I guess we could remove the minions in minions_to_replace from the targets and run `proposal.generate` on the rest.